pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -201,7 +221,9 @@ def module_capsule_configured_mqtt(module_capsule_configured):
     result = module_capsule_configured.execute('firewall-cmd --permanent --add-port="1883/tcp"')
     assert result.status == 0, 'Failed to open mqtt port on capsule'
     module_capsule_configured.execute('firewall-cmd --reload')
-    return module_capsule_configured
+    yield module_capsule_configured
+    if request.config.option.n_minus:
+        raise TypeError('The teardown is missed for MQTT configuration undo for nminus testing')
```

Comment:
TBH We dont need this change but since the setup here without teardown could have impact on feature tests since they are running on same satellite.

Since I am unaware of how the best teardown should happen, I am leaving upto Teams to do it by adding this error. :)

Its a different trick, but the expectation is the person doing the TFA should resolve it !